src/libs/kata-sys-util/src/mount.rs

```diff
@@ -0,0 +1,1029 @@
+    let mut flags: MsFlags = MsFlags::empty();
+    let mut data: Vec<String> = Vec::new();
+
+    for opt in options.iter() {
+        if opt == "defaults" {
+            continue;
+        } else if opt == "loop" {
+            return Err(Error::InvalidMountOption("loop".to_string()));
+        } else if let Some(v) = parse_mount_flags(flags, opt) {
+            flags = v;
+        } else {
+            data.push(opt.clone());
+        }
+    }
+
+    let data = data.join(",");
+    if data.len() > *MAX_MOUNT_PARAM_SIZE {
+        return Err(Error::MountOptionTooBig);
+    }
+
+    Ok((flags, data))
+}
+
+fn parse_mount_flags(mut flags: MsFlags, flag_str: &str) -> Option<MsFlags> {
+    match flag_str {
+        "async" => flags &= !MsFlags::MS_SYNCHRONOUS,
+        "atime" => flags &= !MsFlags::MS_NOATIME,
+        "dev" => flags &= !MsFlags::MS_NODEV,
+        "diratime" => flags &= !MsFlags::MS_NODIRATIME,
+        "exec" => flags &= !MsFlags::MS_NOEXEC,
+        "nomand" => flags &= !MsFlags::MS_MANDLOCK,
+        "norelatime" => flags &= !MsFlags::MS_RELATIME,
+        "nostrictatime" => flags &= !MsFlags::MS_STRICTATIME,
+        "rw" => flags &= !MsFlags::MS_RDONLY,
+        "suid" => flags &= !MsFlags::MS_NOSUID,
+        "bind" => flags |= MsFlags::MS_BIND,
+        "dirsync" => flags |= MsFlags::MS_DIRSYNC,
+        "mand" => flags |= MsFlags::MS_MANDLOCK,
+        "noatime" => flags |= MsFlags::MS_NOATIME,
+        "nodev" => flags |= MsFlags::MS_NODEV,
+        "nodiratime" => flags |= MsFlags::MS_NODIRATIME,
+        "noexec" => flags |= MsFlags::MS_NOEXEC,
+        "nosuid" => flags |= MsFlags::MS_NOSUID,
+        "rbind" => flags |= MsFlags::MS_BIND | MsFlags::MS_REC,
+        "relatime" => flags |= MsFlags::MS_RELATIME,
+        "remount" => flags |= MsFlags::MS_REMOUNT,
+        "ro" => flags |= MsFlags::MS_RDONLY,
+        "strictatime" => flags |= MsFlags::MS_STRICTATIME,
+        "sync" => flags |= MsFlags::MS_SYNCHRONOUS,
+        _ => return None,
+    }
+    Some(flags)
+}
+
+// Do mount, optionally change current working directory if `chdir` is not empty.
+fn mount_at<P: AsRef<Path>>(
+    chdir: Option<PathBuf>,
+    source: P,
+    target: PathBuf,
+    fstype: &str,
+    flags: MsFlags,
+    data: &str,
+) -> Result<()> {
+    let chdir = match chdir {
+        Some(v) => v,
+        None => {
+            return mount(
+                Some(source.as_ref()),
+                &target,
+                Some(fstype),
+                flags,
+                Some(data),
+            )
+            .map_err(|e| Error::Mount(PathBuf::new(), target, e));
+        }
+    };
+
+    info!(
+        sl!(),
+        "mount_at: chdir {}, source {}, target {} , fstype {}, data {}",
+        chdir.display(),
+        source.as_ref().display(),
+        target.display(),
+        fstype,
+        data
+    );
+
+    // TODO: enhance security by `secure_path`.
```

Comment:
Could you create a GH issue and reference it here?

Reply:
https://github.com/kata-containers/kata-containers/issues/3473